drivers/at86rf215/at86rf215_ofdm.c

```diff
@@ -0,0 +1,314 @@
+    at86rf215_reg_write(dev, dev->RF->RG_TXCUTC, RF_PARAMP8U
+                                               | _TXCUTC_LPFCUT(option));
+    /* Set fS; fCUT for TX */
+    at86rf215_reg_write(dev, dev->RF->RG_TXDFE, _TXDFE_SR(option)
+                                              | _TXDFE_RCUT(option));
+
+    /* set channel spacing */
+    at86rf215_reg_write(dev, dev->RF->RG_CS, _channel_spacing_kHz(option) / 25);
+
+    /* set channel center frequency */
+    if (superGHz) {
+        at86rf215_reg_write16(dev, dev->RF->RG_CCF0L, 1 + _channel_center_freq_kHz_2400MHz(option) / 25);
+    } else {
+        at86rf215_reg_write16(dev, dev->RF->RG_CCF0L, 1 + _channel_center_freq_kHz_868MHz(option) / 25);
+    }
+
+    at86rf215_reg_write(dev, dev->BBC->RG_OFDMC, option - 1);
+
+    /* make sure channel config is still valid */
+    dev->num_chans = _get_max_chan(dev, option);
+    dev->netdev.chan = at86rf215_chan_valid(dev, dev->netdev.chan);
+    at86rf215_reg_write16(dev, dev->RF->RG_CNL, dev->netdev.chan);
+}
+
+static unsigned _get_frame_duration(uint8_t option, uint8_t scheme, uint8_t bytes)
+{
+    /* Table 150 - phySymbolsPerOctet values for MR-OFDM PHY, IEEE 802.15.4g-2012 */
+    static const uint8_t quot[] = { 3, 3, 6, 12, 18, 24, 36 };
+    --option;
+    /* phyMaxFrameDuration = phySHRDuration + phyPHRDuration + ceiling [(aMaxPHYPacketSize + 1) x phySymbolsPerOctet] */
+    const unsigned phySHRDuration = OFDM_SYMBOL_TIME_US * 6;
+    const unsigned phyPHRDuration = OFDM_SYMBOL_TIME_US * (option ? 6 : 3);
+    const unsigned phyPDUDuration = (OFDM_SYMBOL_TIME_US * ((bytes + 1) * (1 << option) + quot[scheme] - 1)) / quot[scheme];
+
+    return phySHRDuration + phyPHRDuration + phyPDUDuration;
+}
+
+static void _set_ack_timeout(at86rf215_t *dev, uint8_t option, uint8_t scheme)
+{
+    dev->ack_timeout_usec = dev->csma_backoff_period + IEEE802154G_ATURNAROUNDTIME_US
+                          + _get_frame_duration(option, scheme, AT86RF215_ACK_PSDU_BYTES);
+    DEBUG("[%s] ACK timeout: %"PRIu32" µs\n", "OFDM", dev->ack_timeout_usec);
+}
+
+static bool _option_mcs_valid(uint8_t option, uint8_t mcs)
+{
+    if (option < 1 || option > 4) {
+        return false;
+    }
+
+    if (mcs > BB_MCS_16QAM_3BY4) {
+        return false;
+    }
+
+    if (mcs == BB_MCS_BPSK_REP4 && option > 2) {
+        return false;
+    }
+
+    if (mcs == BB_MCS_BPSK_REP2 && option == 4) {
+        return false;
+    }
+
+    return true;
+}
+
+int at86rf215_configure_OFDM(at86rf215_t *dev, uint8_t option, uint8_t scheme)
+{
+    if (!_option_mcs_valid(option, scheme)) {
+        DEBUG("[%s] invalid option/MCS: %d | %d\n", __func__, option, scheme);
+        return -EINVAL;
+    }
+
+    at86rf215_await_state_end(dev, RF_STATE_TX);
+
+    /* disable radio */
+    at86rf215_reg_write(dev, dev->BBC->RG_PC, 0);
+
+    /* set receiver gain target according to data sheet */
+    at86rf215_reg_write(dev, dev->RF->RG_AGCS, 3 << AGCS_TGT_SHIFT);
```

Comment:
Is that the same?
```suggestion
    at86rf215_reg_write(dev, dev->RF->RG_AGCS, AGCS_TGT_MASK);
```

Reply:
`AGCS_TGT_MASK` is 0xE0, `3 << AGCS_TGT_SHIFT` will work out to 0x60